### PR TITLE
notification issue fixed

### PR DESCRIPTION
**Pull request by mebjas**
Signed-off-by: Minhaz A V <minhazav@gmail.com>

Changes:
`getInitialState` from Notification class and show (or not) the toolbar depending on response. Listener to deal with close dialog persistently modified.

### CODE REVIEW

File: src/components/views/globals/MatrixToolbar.js [hunk 1]
```diff
@@ -32,6 +37,7 @@ module.exports = React.createClass({
     },
 
     render: function() {
+        if (this.state.showToolbar != true) return null;
```
Comment (richvdh) [inline src/components/views/globals/MatrixToolbar.js]: I don't think it's legal to return `null` here. But anyway, this logic is in the wrong place; the logic for whether to show the toolbar or not belongs in the parent component (MatrixChat).

`MatrixChat.render` already has: 
```
... if (Notifier.supportsDesktopNotifications() && !Notifier.isEnabled() && !Notifier.isToolbarHidden()) {
            topBar = <MatrixToolbar />;
}
```

You just need to update `Notifier.isToolbarHidden`.

File: src/components/views/globals/MatrixToolbar.js [hunk 2]
```diff
@@ -18,13 +18,18 @@ limitations under the License.
...
 module.exports = React.createClass({
     displayName: 'MatrixToolbar',
 
+    getInitialState: function() {
+        return {showToolbar: Notifier.showToolbar()};
+    },
+
     hideToolbar: function() {
-        Notifier.setToolbarHidden(true);
+        // TODO (mebjas): may be a add a confirm dialog here @tag: ara4n
```
Comment (richvdh) [inline src/components/views/globals/MatrixToolbar.js]: I think a confirm dialog would be overkill

File: src/components/views/globals/MatrixToolbar.js (hunk 1 above)
Comment (mebjas) [inline src/components/views/globals/MatrixToolbar.js]: Cool, though if the user clicked on block in the notification permission dialog, we are going to show `you have disabled desktop notification` right? The logic for that would come here right?

**Comment by mebjas** [comment]
Well with that all logic moves to `react-sdk` I'll close this pull.

**Pull request closed without merging**